src/components/IconChooser.js

```diff
@@ -42,7 +41,7 @@ export default class IconChooser extends Component {
           }
         </View>
         <Button
-          onPress={this.refreshIcons}
+          onPress={() => refreshIcons(this)}
```

Comment:
This has much worse performance since with every `render` the callback function is different, keeping the function in the component is recommended. (or writing another wrapper if it's not possible).